pkg/virt-launcher/virtwrap/network/common.go

```diff
@@ -328,38 +328,26 @@ func (h *NetworkUtilsHandler) GetMacDetails(iface string) (net.HardwareAddr, err
 }
 
 // SetRandomMac changes the MAC address for a given interface to a randomly generated, preserving the vendor prefix
-func (h *NetworkUtilsHandler) SetRandomMac(iface string) (net.HardwareAddr, error) {
-	var mac net.HardwareAddr
-
-	currentMac, err := h.GetMacDetails(iface)
-	if err != nil {
-		return nil, err
-	}
-
-	changed := false
-
+func (h *NetworkUtilsHandler) SetRandomMac(ifaceName string) (*net.HardwareAddr, error) {
 	for i := 0; i < randomMacGenerationAttempts; i++ {
-		changed, err = lmf.SpoofMacSameVendor(iface, false)
+		changed, err := lmf.SpoofMacSameVendor(ifaceName, false)
 		if err != nil {
-			log.Log.Reason(err).Errorf("failed to spoof MAC for an interface: %s", iface)
+			log.Log.Reason(err).Errorf("failed to spoof MAC for an interface: %s", ifaceName)
 			return nil, err
 		}
 
 		if changed {
-			mac, err = h.GetMacDetails(iface)
+			updatedIface, err := h.LinkByName(ifaceName)
 			if err != nil {
 				return nil, err
```

Comment:
I'm unsure if we really do want to blow the pod up here if we fail to read the new mac.

Reply:
@maiqueb OK how about taking this check out of the `SetRandomMac` function, and create a new one called `SafeSetRandomMac` which will wrap the original one but it will also do the check with `h.LinkByName`